src/node/internal/internal_dns_client.ts

```diff
@@ -71,6 +71,11 @@ export async function sendDnsRequest(
       },
       method: 'GET',
     });
+    if (!response.ok) {
+      throw new Error(
+        `response not ok ${String(response.status)} ${response.statusText}`
```

Comment:
@anonrig a DNSError will be thrown in the catch below, whatever we throw here doesn't really matter.

Does that sound ok?

Reply:
Or maybe we should throw a BAD_RESP instead. The error is visible to the user, right?

Reply:
This error is visible to the user yes. I suppose BAD_RESP is a better error code.